tools/c7n_azure/c7n_azure/resources/front_door.py

```diff
@@ -72,3 +73,22 @@ def process(self, resources, event=None):
                 if self.check_state(front_endpoint.web_application_firewall_policy_link):
                     matched.append(front_door)
         return matched
+
+
+@FrontDoor.filter_registry.register('web-application-firewall-policies')
```

Comment:
rename to firewall-policy, this would be better to derive as a list item filter afaics.

Reply:
As far as i understood one Front Door resource can have multiple frontend endpoints whereas each individual endpoint can optionally have one WAF Policy. So, in general, one Front Door can have multiple different WAF policies (if it has multiple frontend endpoints). I rewrote this filter deriving from list-item filter. But this implementation neglects the connection between each policy and its frontend endpoint. Not sure whether it's okey..

Reply:
Also, regarding `front-door.waf` filter, seems like the list returned from `process()` method can contain duplicated Front Door resources in case some Front Door have multiple enabled frontend endpoints.